lib/install_go.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+# Borrowing this script from the Ruby Buildpack.  
+# https://raw.githubusercontent.com/cloudfoundry/nodejs-buildpack/master/scripts/install_go.sh
+
+set -euo pipefail
+
+GO_VERSION="1.11.5"
+
+if [ $CF_STACK == "cflinuxfs2" ]; then
+    GO_SHA256="51cab63f3de5e2f75a9036801712e4d7ae9bf226f0b61abce8d784e698148d3b"
+elif [ $CF_STACK == "cflinuxfs3" ]; then
+    GO_SHA256="ee770df4e1863ee8e07574cb48e0245b61bec8f118faf6ec3742ea89eb20db28"
+else
+  echo "       **ERROR** Unsupported stack"
+  echo "                 See https://docs.cloudfoundry.org/devguide/deploy-apps/stacks.html for more info"
+  exit 1
+fi
+
+export GoInstallDir="/tmp/go$GO_VERSION"
+mkdir -p $GoInstallDir
+
+if [ ! -f $GoInstallDir/go/bin/go ]; then
+  URL=https://buildpacks.cloudfoundry.org/dependencies/go/go${GO_VERSION}.linux-amd64-${CF_STACK}-${GO_SHA256:0:8}.tar.gz
+
+  echo "-----> Download go ${GO_VERSION}"
+  curl -s -L --retry 15 --retry-delay 2 $URL -o /tmp/go.tar.gz
+
+  DOWNLOAD_SHA256=$(shasum -a 256 /tmp/go.tar.gz | cut -d ' ' -f 1)
+
+  if [[ $DOWNLOAD_SHA256 != $GO_SHA256 ]]; then
+    echo "       **ERROR** SHA256 mismatch: got $DOWNLOAD_SHA256 expected $GO_SHA256"
+    exit 1
+  fi
+
+  tar xzf /tmp/go.tar.gz -C $GoInstallDir
+  rm /tmp/go.tar.gz
+fi
+if [ ! -f $GoInstallDir/go/bin/go ]; then
+  echo "       **ERROR** Could not download go"
+  exit 1
+fi
```

Comment:
Would you mind adding the newline to this script?

```suggestion
fi

```

Reply:
fixed on next commit.